core/kazoo_apps/src/kapps_util.erl

```diff
@@ -418,17 +414,62 @@ do_get_ccvs_by_ip(IP) ->
 
 -spec account_ccvs_from_ip_auth(kz_json:object()) -> kz_proplist().
 account_ccvs_from_ip_auth(Doc) ->
-    AccountID = kz_json:get_value([<<"value">>, <<"account_id">>], Doc),
-    OwnerID = kz_json:get_value([<<"value">>, <<"owner_id">>], Doc),
+    AccountId = kz_json:get_value([<<"value">>, <<"account_id">>], Doc),
+    OwnerId = kz_json:get_value([<<"value">>, <<"owner_id">>], Doc),
     AuthType = kz_json:get_value([<<"value">>, <<"authorizing_type">>], Doc, <<"anonymous">>),
 
-    props:filter_undefined(
-      [{<<"Account-ID">>, AccountID}
-      ,{<<"Owner-ID">>, OwnerID}
-      ,{<<"Authorizing-ID">>, kz_doc:id(Doc)}
-      ,{<<"Inception">>, <<"on-net">>}
-      ,{<<"Authorizing-Type">>, AuthType}
-      ]).
+    case are_all_enabled([{<<"account">>, AccountId}
+                         ,{<<"owner">>, OwnerId}
+                         ,{AuthType, kz_doc:id(Doc)}
+                         ])
+    of
+        {'false', _Reason} ->
+            lager:notice("no IP auth info: ~p", [_Reason]),
+            [];
+        'true' ->
+            props:filter_undefined(
+              [{<<"Account-ID">>, AccountId}
+              ,{<<"Owner-ID">>, OwnerId}
+              ,{<<"Authorizing-ID">>, kz_doc:id(Doc)}
+              ,{<<"Inception">>, <<"on-net">>}
+              ,{<<"Authorizing-Type">>, AuthType}
+              ])
+    end.
+
+-type not_enabled_error() :: 'device_disabled' |
+                             'owner_disabled' |
+                             'account_disabled'.
+-spec are_all_enabled(kz_proplist()) ->
+                             'true' |
+                             {'false', {not_enabled_error(), ne_binary()}}.
+are_all_enabled(Things) ->
+    AccountId = props:get_value(<<"account">>, Things),
```

Comment:
How about `?MATCH_ACCOUNT_RAW(AccountId) = ...` to make sure this is never `undefined`?